[
    {
        "content": "<p><a href=\"https://github.com/brettcannon\">brettcannon</a> added the bug label to <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">Issue #7829</a>.</p>",
        "id": 418346734,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1706300831
    },
    {
        "content": "<p>brettcannon opened <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<h3>Test Case</h3>\n<p>TODO: upload Wasm file here</p>\n<h3>Steps to Reproduce</h3>\n<ul>\n<li><a href=\"https://github.com/python/cpython/blob/07236f5b39a2e534cf190cd4f7c73300d209520b/Lib/test/test_posix.py#L452-L454\">https://github.com/python/cpython/blob/07236f5b39a2e534cf190cd4f7c73300d209520b/Lib/test/test_posix.py#L452-L454</a></li>\n</ul>\n<h3>Expected Results</h3>\n<p>Test should pass (it does under preview 1).</p>\n<h3>Actual Results</h3>\n<p>Raises errno 63: <a href=\"https://docs.python.org/3/library/errno.html#errno.ENOSR\"><code>ENOSR</code></a> (out of stream resources).</p>\n<h3>Versions and Environment</h3>\n<p>Wasmtime version or commit: 17.0.0</p>\n<p>Operating system: Linux/WSL</p>\n<p>Architecture: x64</p>\n<h3>Extra Info</h3>\n<p>Anything else you'd like to add?</p>\n<p>Success logs under preview 1: <br>\n<a href=\"https://github.com/bytecodealliance/wasmtime/files/14069110/success-utime-fd.txt\">success-utime-fd.txt</a></p>\n<p>Failure logs under preview 2:<br>\n<a href=\"https://github.com/bytecodealliance/wasmtime/files/14069114/failure-utime-fd.txt\">failure-utime-fd.txt</a><br>\n</p>\n</blockquote>",
        "id": 418346736,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1706300832
    },
    {
        "content": "<p>alexcrichton <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1912743404\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p>Could you perhaps reupload the failure logs? It currently seems to be empty and/or says success with no extra logs</p>\n</blockquote>",
        "id": 418356242,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1706305925
    },
    {
        "content": "<p>brettcannon <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1912825812\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p><a href=\"https://github.com/bytecodealliance/wasmtime/files/14070134/failure-utime-fd.txt\">failure-utime-fd.txt</a></p>\n<p>Sorry about that!</p>\n</blockquote>",
        "id": 418365697,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1706311890
    },
    {
        "content": "<p>pchickey <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1928579086\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p>@elliottt and I investigated this and will be posting a fix shortly.</p>\n</blockquote>",
        "id": 419955165,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707180717
    },
    {
        "content": "<p>pchickey <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1928598422\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p>@brettcannon We read the test source you linked to and believed we reproduced it in PR #7876 with a modification to one of our existing tests, and then fixed it in the same PR. I don't know how to run your tests locally to verify that this fixed it, but we think we got the underlying bug in our implementation fixed now. Please verify for us whether this is a fix when you have time, and also if it changes the status of any other of your posix tests, which exercise a lot more than our test suite.</p>\n</blockquote>",
        "id": 419957346,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707182115
    },
    {
        "content": "<p>brettcannon <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1930550657\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<blockquote>\n<p>I don't know how to run your tests locally to verify that this fixed it</p>\n</blockquote>\n<p>FYI I'm working on better docs for this, but <a href=\"https://github.com/python/cpython/tree/main/Tools/wasm#wasi-wasm32-wasi\">https://github.com/python/cpython/tree/main/Tools/wasm#wasi-wasm32-wasi</a> are the current docs if you're interested.</p>\n<blockquote>\n<p>Please verify for us whether this is a fix when you have time</p>\n</blockquote>\n<p>Will do! I'll plan to give it a try this week.</p>\n<blockquote>\n<p>if it changes the status of any other of your posix tests, which exercise a lot more than our test suite.</p>\n</blockquote>\n<p>Yeah, Python's test suite is old and extensive for this sort of thing: we're great at exercising C compilers and OSs. <span aria-label=\"grinning face with smiling eyes\" class=\"emoji emoji-1f601\" role=\"img\" title=\"grinning face with smiling eyes\">:grinning_face_with_smiling_eyes:</span> If you're ever interested in trying to get the Python test suite set up as a 3rd-party test suite to run, just let me know.</p>\n</blockquote>",
        "id": 420111867,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707245128
    },
    {
        "content": "<p>brettcannon <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1936792577\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p>Commented at <a href=\"https://github.com/bytecodealliance/wasmtime/pull/7876#issuecomment-1936792352\">https://github.com/bytecodealliance/wasmtime/pull/7876#issuecomment-1936792352</a> about the PR passing CPython's test suite.</p>\n</blockquote>",
        "id": 420758107,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707528024
    },
    {
        "content": "<p>pchickey closed <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<h3>Test Case</h3>\n<p>TODO: upload Wasm file here</p>\n<h3>Steps to Reproduce</h3>\n<ul>\n<li><a href=\"https://github.com/python/cpython/blob/07236f5b39a2e534cf190cd4f7c73300d209520b/Lib/test/test_posix.py#L452-L454\">https://github.com/python/cpython/blob/07236f5b39a2e534cf190cd4f7c73300d209520b/Lib/test/test_posix.py#L452-L454</a></li>\n</ul>\n<h3>Expected Results</h3>\n<p>Test should pass (it does under preview 1).</p>\n<h3>Actual Results</h3>\n<p>Raises errno 63: <a href=\"https://docs.python.org/3/library/errno.html#errno.ENOSR\"><code>ENOSR</code></a> (out of stream resources).</p>\n<h3>Versions and Environment</h3>\n<p>Wasmtime version or commit: 17.0.0</p>\n<p>Operating system: Linux/WSL</p>\n<p>Architecture: x64</p>\n<h3>Extra Info</h3>\n<p>Anything else you'd like to add?</p>\n<p>Success logs under preview 1: <br>\n<a href=\"https://github.com/bytecodealliance/wasmtime/files/14069110/success-utime-fd.txt\">success-utime-fd.txt</a></p>\n<p>Failure logs under preview 2:<br>\n<a href=\"https://github.com/bytecodealliance/wasmtime/files/14069114/failure-utime-fd.txt\">failure-utime-fd.txt</a><br>\n</p>\n</blockquote>",
        "id": 421534078,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707945331
    },
    {
        "content": "<p>pchickey <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829#issuecomment-1944615597\">commented</a> on <a href=\"https://github.com/bytecodealliance/wasmtime/issues/7829\">issue #7829</a>:</p>\n<blockquote>\n<p>This landed in <code>main</code> and in <code>release-18.0.0</code> so I'm going to close it now. We chose not to release this fix as a patch release on <code>release-17.0.0</code> but if someone really needs it before 18 goes out in 7 days, we still could.</p>\n</blockquote>",
        "id": 421534079,
        "sender_full_name": "Wasmtime GitHub notifications bot",
        "timestamp": 1707945332
    }
]